zerver/lib/markdown/__init__.py

```diff
@@ -690,7 +689,7 @@ def add_embed(self, root: Element, link: str, extracted_data: UrlEmbedData) -> N
 
         img_link = get_camo_url(extracted_data.image)
         img = SubElement(container, "a")
-        img.set("style", "background-image: url(" + css_escape(img_link) + ")")
+        img.set("style", 'background-image: url("' + re.sub(r'(\n|\\|")', r"\\\1", img_link) + '")')
```

Comment:
In a CSS quoted string, the backslash-newline sequence represents nothing, not a newline; a newline needs to be escaped as `\a ` (backslash-a-space).

```suggestion
        img.set(
            "style",
            'background-image: url("'
            + img_link.replace("\\", "\\\\").replace('"', '\\"').replace("\n", "\\a ")
            + '")',
        )
```

(My benchmark indicates `.replace` is slightly faster than `.translate` or `re.sub` or here.)

Reply:
Sounds good -- applied!